### PR TITLE
distribute libcore.a and run demo.cpp. 

### PR DESCRIPTION
**Pull request by T8T9**
1. distribute libcore.a and related header files and third party lib and header files to build/dist directory.
2. run demo.cpp when execute `./build.sh ci`

### CODE REVIEW

File: tutorials/demo.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#include <iostream>
+
+#include "cinn/backends/compiler.h"
+#include "cinn/cinn.h"
+#include "cinn/common/test_helper.h"
```
Comment (Superjomn) [inline tutorials/demo.cpp]: C++ source 统一用 .cc 后缀吧

File: cinn/common/CMakeLists.txt [hunk 2]
```diff
@@ -17,12 +17,37 @@ set(srcs
     union_find.cc
     )
 
+set(includes
+    arithmatic.h
+    cas.h
+    common.h
+    cuda_test_helper.h
```
Comment (Superjomn) [inline cinn/common/CMakeLists.txt]: 直接 GLOB 把所有头文件加进去？ 应该没啥副作用

File: tutorials/demo.cpp (hunk 1 above)
Comment (T8T9) [reply tutorials/demo.cpp]: done

File: cinn/common/CMakeLists.txt (hunk 2 above)
Comment (T8T9) [reply cinn/common/CMakeLists.txt]: done

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**